host/sgx/create.c

```diff
@@ -849,6 +859,16 @@ oe_result_t oe_sgx_build_enclave(
     /* Validate the enclave prop_override structure */
     OE_CHECK(oe_sgx_validate_enclave_properties(&props, NULL));
 
+    if (enclave->simulate && context->create_zero_base_enclave)
+    {
+        OE_TRACE_WARNING(
+            "Requested creation of 0-base enclave in simulation mode, "
+            "which is currently not supported. Ignoring request to "
+            "create 0-base enclave.\n");
+        context->create_zero_base_enclave = 0;
```

Comment:
Do we need to clear create_zero_base_enclave and start_address here? They're set in line 929-932 anyway. Can you double check?

Reply:
I am changing this back to fail with OE_INVALID_PARAMETER when 0-base is requested in simulation mode. Need to explore more to downgrade it to normal enclave. Created issue #4134 

Current behavior:

/home/ratrames/dev/OE_0_base/tests/sim-mode/host/host.c(22): error: oe_create_sim_mode_enclave(): result=3
2: 2021-07-01T01:24:56+0000.410393Z [(H)ERROR] tid(0x7fdfe77bfb80) | Requested creation of 0-base enclave in simulation mode, which is currently not supported.
2:  [/home/ratrames/dev/OE_0_base/host/sgx/create.c:oe_sgx_build_enclave:929]
2: 2021-07-01T01:24:56+0000.410518Z [(H)ERROR] tid(0x7fdfe77bfb80) | :OE_INVALID_PARAMETER [/home/ratrames/dev/OE_0_base/host/sgx/create.c:oe_create_enclave:1208]